infrastructure/microvm/cloudhypervisor/state.go

```diff
@@ -0,0 +1,116 @@
+package cloudhypervisor
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"github.com/spf13/afero"
+	"github.com/weaveworks-liquidmetal/flintlock/core/models"
+	"github.com/weaveworks-liquidmetal/flintlock/pkg/defaults"
+)
+
+type State interface {
+	Root() string
+
+	PID() (int, error)
+	PIDPath() string
+	SetPid(pid int) error
+
+	LogPath() string
+	StdoutPath() string
+	StderrPath() string
+	SockPath() string
+
+	//MetadataDir() string
+	CloudInitImage() string
+}
+
+func NewState(vmid models.VMID, stateDir string, fs afero.Fs) State {
+	return &fsState{
+		stateRoot: fmt.Sprintf("%s/%s", stateDir, vmid.String()),
+		fs:        fs,
+	}
+}
+
+type fsState struct {
+	stateRoot string
+	fs        afero.Fs
+}
+
+func (s *fsState) Root() string {
+	return s.stateRoot
+}
+
+func (s *fsState) PIDPath() string {
+	return fmt.Sprintf("%s/cloudhypervisor.pid", s.stateRoot)
+}
+
+func (s *fsState) PID() (int, error) {
+	return s.pidReadFromFile(s.PIDPath())
+}
+
+func (s *fsState) LogPath() string {
+	return fmt.Sprintf("%s/cloudhypervisor.log", s.stateRoot)
+}
+
+func (s *fsState) StdoutPath() string {
+	return fmt.Sprintf("%s/cloudhypervisor.stdout", s.stateRoot)
+}
+
+func (s *fsState) StderrPath() string {
+	return fmt.Sprintf("%s/cloudhypervisor.stderr", s.stateRoot)
+}
+
+func (s *fsState) SockPath() string {
+	return fmt.Sprintf("%s/cloudhypervisor.sock", s.stateRoot)
+}
+
+// func (s *fsState) MetadataDir() string {
+// 	return fmt.Sprintf("%s/metadata", s.stateRoot)
+// }
+
+func (s *fsState) CloudInitImage() string {
+	return fmt.Sprintf("%s/cloud-init.img", s.stateRoot)
+}
+
+func (s *fsState) SetPid(pid int) error {
+	return s.pidWriteToFile(pid, s.PIDPath())
+}
+
+func (s *fsState) pidReadFromFile(pidFile string) (int, error) {
+	file, err := s.fs.Open(pidFile)
+	if err != nil {
+		return -1, fmt.Errorf("opening pid file %s: %w", pidFile, err)
```

Comment:
I think it is more done to return the nil value for a go int which is `0`... but i know kernel syscalls have `-1` as theirs for file funcs soooo 🤷‍♀️

Reply:
We use `-1` in the firecracker side. Lets keep as `-1` for the time being, we can always change. 

I have moved the pid code to shared though now